### PR TITLE
docs(gatsby-plugin-guess-js): Document JWT

### PR DESCRIPTION
**Pull request by omonk**
## Description

`gatsby-plugin-guess-js` is able to accept a JWT for authenticating the plugin on deployment services. 

The documentation previously didn't have any mention about how to use this on deployment services as locally it requires human interaction to grant permissions.

## Related Issues

https://github.com/guess-js/gatsby-guess/issues/19#issuecomment-440305187
https://github.com/gatsbyjs/gatsby/issues/15582

### CODE REVIEW

File: packages/gatsby-plugin-guess-js/README.md [hunk 1]
```diff
@@ -36,6 +36,8 @@ module.exports = {
       options: {
         // Find the view id in the GA admin in a section labeled "views"
         GAViewID: `VIEW_ID`,
+        // Add JWT token to perform authentication on deployment builds
+        jwt: GA_JWT
```
Comment (wardpeet) [inline packages/gatsby-plugin-guess-js/README.md]: Could you explain what GA_JWT holds?

For gatsbyjs.org we're doing the following to make it work:
```
jwt: {
  client_email: process.env.ANALYTICS_SERVICE_ACCOUNT,
  private_key: process.env.ANALYTICS_SERVICE_ACCOUNT_KEY,
},
```

ENV vars look like:
```
ANALYTICS_SERVICE_ACCOUNT="<name>@<project>.iam.gserviceaccount.com"
```
```
ANALYTICS_SERVICE_ACCOUNT_KEY="-----BEGIN PRIVATE KEY-----\nPRIVATE KEY STRING\n-----END PRIVATE KEY-----"
```

File: packages/gatsby-plugin-guess-js/README.md [hunk 2]
```diff
@@ -47,3 +49,15 @@ module.exports = {
...
+
+Using the plugin locally you'll need to authenticate the plugin with GA to gain access to your data. When building you'll be required to sign in to the organisation you'll be feeding the data from.
+
+### Production
+
+If deploying with a service such as netlify or heroku etc, you wont be able to grant access via Google Sign On. To get around this you'll need to generate a JWT for your GA account to pass to the plugin, this will enable the plugin to work in prod without any human interaction to authenticate permissions.
+
+## How to get a JWT token?
+
+[Here](https://2ality.com/2015/10/google-analytics-api.html)
```
Comment (wardpeet) [inline packages/gatsby-plugin-guess-js/README.md]: I feel like we should just copy and modify https://2ality.com/2015/10/google-analytics-api.html#unlocking-the-google-analytics-api in the readme and give credits to 2ality url.

File: packages/gatsby-plugin-guess-js/README.md (hunk 1 above)
Comment (omonk) [reply packages/gatsby-plugin-guess-js/README.md]: very good point, tried to follow the rabbit hole into Guess webpack and ga packages but haven't been able to work out where this goes yet. I can see that client_email, private_key combo in `www/` but it appears to be commented out currently?

File: packages/gatsby-plugin-guess-js/README.md (hunk 2 above)
Comment (omonk) [reply packages/gatsby-plugin-guess-js/README.md]: Another good point, I would probably be welcome to more suggestions as to how to do this in a simpler way (if possible?).

Maybe attaching something to a gatsby preBuild api to get a fresh JWT. My knowledge on JWT is a little limited but I would assume that this wouldn't be the best way as each JWT is valid until you make it non valid correct?

**Comment by wardpeet** [comment]
My mood changes fast :joy:. Let's re-open this one and wait for https://github.com/gatsbyjs/gatsby/pull/15774 to get merged and get some of your updates in as well. So you can get credit for your work! 💪

**Comment by omonk** [comment]
🙈 totally missed that! Very similar to my changes yep. I think it would be really beneficially to add a solution to get the JWT from google in there as well!

**Comment by wardpeet** [comment]
I would copy the first step of https://2ality.com/2015/10/google-analytics-api.html. Add a small paragraph that the token will expire in x time. If you need to update your token automatically please read https://2ality.com/2015/10/google-analytics-api.html

That's how I would suggest it 🤷‍♂

**Comment by sidharthachatterjee** [comment]
@omonk I've made the changes @wardpeet requested but don't have access to push to your fork. Can you please allow me write access to your fork? I'll push the changes and get this merged in ASAP

**Comment by omonk** [comment]
@sidharthachatterjee Sorry for my delayed response, have been on holiday!

Thanks very much for merging that!

**Comment by wardpeet** [comment]
I hope you had an awesome vacation! 🌴 Thanks for your work!

**Pull request merged**